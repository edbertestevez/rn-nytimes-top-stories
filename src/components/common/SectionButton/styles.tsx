import styled from 'styled-components/native';
import {ISectionProps} from '.';
import {defaultFontSize} from '../../../styles/font';
import {defaultTheme} from '../../../styles/theme';

export const ButtonContainer = styled.TouchableOpacity`
  background-color: ${defaultTheme.PRIMARY_BUTTON_COLOR};
  padding-horizontal: 12px;
  border-width: 1px;
  width: ${(props: ISectionProps) => props.width || 160}
  height: ${(props: ISectionProps) => props.height || 40}
  border-color: ${(props: ISectionProps) =>
    props.selected
      ? defaultTheme.SELECTED_COLOR
      : defaultTheme.DEFAULT_BORDER_COLOR};
  align-items: center;
  justify-content: center;
  margin-right: 16;
  margin-bottom: 8;
  border-radius: 8;
`;

export const ButtonText = styled.Text`
  color: ${(props: ISectionProps) =>
    props.selected
      ? defaultTheme.SELECTED_COLOR
      : defaultTheme.PRIMARY_TEXT_COLOR};
  font-weight: ${(props: ISectionProps) => (props.selected ? 700 : 'normal')}
  font-size: ${defaultFontSize.MEDIUM};
`;
