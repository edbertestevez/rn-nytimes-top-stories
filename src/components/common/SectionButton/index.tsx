import React from 'react';
import {DEFAULT_TOUCHABLE_OPACITY} from '../../../config/widgets';
import {ButtonContainer, ButtonText} from './styles';

export interface ISectionProps {
  width?: number;
  height?: number;
  selected: boolean;
}

interface IProps extends ISectionProps {
  label?: string;
  onPress: () => void;
}

const SectionButton: React.FC<IProps> = (props) => {
  let {label, selected} = props;

  return (
    <ButtonContainer activeOpacity={DEFAULT_TOUCHABLE_OPACITY} {...props}>
      <ButtonText selected={selected}>{label}</ButtonText>
    </ButtonContainer>
  );
};

export default SectionButton;
