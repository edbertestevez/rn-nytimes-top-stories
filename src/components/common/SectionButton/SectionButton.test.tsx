import 'jsdom-global/register';
import 'jest-styled-components';

import React from 'react';
import 'react-native';
import Enzyme, {mount} from 'enzyme';
import SectionButton from './index';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({adapter: new Adapter()});

describe('<SectionButton />', () => {
  const DEFAULT_LABEL = 'Test';
  const DEFAULT_FUNCTION = () => console.log(DEFAULT_LABEL);

  const testComponent = (
    <SectionButton
      label={DEFAULT_LABEL}
      onPress={DEFAULT_FUNCTION}
      selected={true}
    />
  );

  const wrapper = mount(testComponent);
  const tree = renderer.create(testComponent).toJSON();
  it('renders correctly', () => {
    expect(tree).toMatchSnapshot();
  });
  it('renders correct button label', () => {
    expect(wrapper.props().label).toEqual(DEFAULT_LABEL);
  });
});
