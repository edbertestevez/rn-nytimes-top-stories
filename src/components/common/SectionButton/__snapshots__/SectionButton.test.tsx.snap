// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionButton /> renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#918AE4",
      "borderRadius": 8,
      "borderWidth": 1,
      "height": 40,
      "justifyContent": "center",
      "marginBottom": 8,
      "marginRight": 16,
      "opacity": 1,
      "paddingHorizontal": 12,
      "width": 160,
    }
  }
>
  <Text
    selected={true}
    style={
      Array [
        Object {
          "color": "#918AE4",
          "fontSize": 16,
          "fontWeight": "700",
        },
      ]
    }
  >
    Test
  </Text>
</View>
`;
